docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
 
-    let zs' =
-        seq {
-            for i in 0..9 do
-                printfn "Adding %d" i
-                yield i
-        }
+    // Construction
+    let numberAndWord = (1, "Hello")
+    let numberAndWordAndNow = (1, "Hello", System.DateTime.Now)
 
-<div id="tuples-and-records"></div>
+    // Deconstruction
+    let (number, word) = numberAndWord
+    let (_, _, now) = numberAndWordAndNow
 
-## Tuples and Records
+    // fst and snd functions for two-item tuples:
+    let number = fst numberAndWord
+    let word = snd numberAndWord
 
-### Tuple
-A *tuple* is a grouping of unnamed but ordered values, possibly of different types:
+    // Pattern matching
+    let printNumberAndWord numberAndWord =
+        match numberAndWord with
+        | (1, word) -> printfn $"One: %s{word}"
+        | (2, word) -> printfn $"Two: %s{word}"
+        | (_, word) -> printfn $"Number: %s{word}"
 
-    // Tuple construction
-    let x = (1, "Hello")
+    // Function parameter deconstruction
+    let printNumberAndWord' (number, word) = printfn $"%d{number}: %s{word}"
 
-    // Triple
-    let y = ("one", "two", "three")
+In C#, if a method has an `out` parameter (e.g. [`DateTime.TryParse`](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tryparse)) the `out` result will be part of a tuple.
 
-    // Tuple deconstruction / pattern
-    let (a', b') = x
+    let (success, outParsedDateTime) = System.DateTime.TryParse("2001/02/06")
 
-The first and second elements of a tuple can be obtained using `fst`, `snd`, or pattern matching:
+See [Tuples (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/tuples) for learn more.
 
-    let c' = fst (1, 2)
-    let d' = snd (1, 2)
+<div id="data-types-records"></div>
 
-    let print' tuple =
-        match tuple with
-        | (a, b) -> printfn "Pair %A %A" a b
+### Records
 
-### Record
-*Records* represent simple aggregates of named values, optionally with members:
+*Records* represent aggregates of named values. They are sealed classes with extra toppings: default immutability, structural equality, and pattern matching support.
 
-    // Declare a record type
-    type Person = { Name : string; Age : int }
+    // Declare
+    type Person = { Name: string; Age: int }
+    type Car =
+        { Make: string
+          Model: string
+          Year: int }
 
-    // Create a value via record expression
+    // Create
     let paul = { Name = "Paul"; Age = 28 }
 
-    // 'Copy and update' record expression
+    // Duplicate
     let paulsTwin = { paul with Name = "Jim" }
 
-Records can be augmented with properties and methods:
-
-    type Person with
-        member x.Info = (x.Name, x.Age)
-
-Records are essentially sealed classes with extra topping: default immutability, structural equality, and pattern matching support.
+    // Built-in equality
+    let evilPaul = { Name = "Paul"; Age = 28 }
+    paul = evilPaul  // true
 
+    // Pattern matching
```

Comment:
have I ever mentioned I love `function` :D (not sure where it should be covered)

Reply:
I'm going to start looking at a `cheat-sheet-short` soon.